File: apps/minecraft/velocity/src/test/java/crabcraft/net/crabUtilities/velocity/JadeClientProtocolPayloadRegressionTest.java
package crabcraft.net.crabUtilities.velocity;

import java.util.Arrays;

final class JadeClientProtocolPayloadRegressionTest {

    public static void main(String[] args) {
        byte[] encoded = ConnectionListener.encodeClientProtocol(776);
        check(Arrays.equals(encoded, new byte[]{0, 0, 3, 8}),
                "the proxy encoded the 26.2 protocol payload incorrectly");
    }

    private static void check(boolean condition, String message) {
        if (!condition) throw new AssertionError(message);
    }
}

File: apps/minecraft/velocity/src/test/kotlin/crabcraft/net/crabUtilities/velocity/JadeClientProtocolPayloadRegressionTest.kt
package crabcraft.net.crabUtilities.velocity

object JadeClientProtocolPayloadRegressionTest {
    @JvmStatic
    fun main(args: Array<String>) {
        val encoded = ConnectionListener.encodeClientProtocol(776)
        if (!encoded.contentEquals(byteArrayOf(0, 0, 3, 8))) {
            throw AssertionError("the proxy encoded the 26.2 protocol payload incorrectly")
        }
    }
}
